File: dcp-solutions/2024-11-06/solution.py
```python
def break_string(s, k):
    words = s.split()
    result = []
    current_line = ""
    
    for word in words:
        if len(current_line) + len(word) + 1 > k:
            # not enough space to put this word on the current line,
            # so start a new line
            if not current_line:
                return None
            result.append(current_line)
            current_line = word
        else:
            # can fit the word on the current line
            if current_line:
                current_line += " "
            current_line += word
    
    # add the last line to the result
    if current_line:
        result.append(current_line)
    
    return result
```

File: dcp-solutions/2024-11-06/solution.js
```javascript
function breakString(s, k) {
    let words = s.split(' ');
    let result = [];
    let currentLine = '';
    
    for (let word of words) {
        if (currentLine.length + word.length > k) {
            if (!currentLine) continue;
            result.push(currentLine);
            currentLine = word;
        } else {
            if (currentLine) currentLine += ' ';
            currentLine += word;
        }
    }

    if (currentLine) result.push(currentLine);

    return result.length === 0 ? null : result;
}
```